src/Plugin/rest/resource/GlobalMobileMenu.php

```diff
@@ -0,0 +1,186 @@
+      $plugin_definition
+    );
+    $instance->configFactory = $container->get('config.factory');
+    $instance->languageManager = $container->get('language_manager');
+    $instance->apiManager = $container->get('helfi_navigation.api_manager');
+    $instance->environmentResolver = $container->get('helfi_api_base.environment_resolver');
+    $instance->localMenuTreeBuilder = $container->get('helfi_navigation.menu_tree_builder');
+
+    return $instance;
+  }
+
+  /**
+   * Callback for GET requests.
+   *
+   * @return \Drupal\rest\ResourceResponse
+   *   The response.
+   */
+  public function get(): ResourceResponse {
+    $langcode = $this->languageManager
+      ->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)
+      ->getId();
+
+    try {
+      $apiResponse = $this->apiManager->get($langcode, 'main', []);
+    }
+    catch (\Exception $exception) {
+      return new ResourceResponse([], 404);
+    }
+
+    // If authorization key is set, just return the menu without enrichment.
+    if ($this->apiManager->isAuthorized()) {
+      return new ResourceResponse(json_decode(json_encode($apiResponse->data), TRUE), 200);
+    }
+
+    $environment = $this->environmentResolver->getActiveEnvironment();
+    $site_name = $this->configFactory->get('system.site')->get('name');
+    $project_name = $environment->getId();
+
+    // Create menu tree and add data to the local menu.
+    $menuTree = $this->getTreeFromMainMenu();
+    $menuTree['is_injected'] = TRUE;
+
+    // Commented lines are present in the api request,
+    // most likely generated by the drupal api module.
+    $site_data = (object) [
+      // 'content_translation_changed' => '',
+      // 'content_translation_created' => '',
+      // 'content_translation_created' => '',
+      // 'content_translation_source' => '',
+      // 'content_translation_uid' => '',
+      // 'metatag' => '',
+      // 'default_langcode' => [(object) ['value' => TRUE]],
+      'langcode' => [(object) ['value' => $langcode]],
+      'menu_tree' => [0 => (object) $menuTree],
+      'name' => [(object) ['value' => $site_name]],
+      'project' => [(object) ['value' => $project_name]],
+      'status' => [(object) ['value' => TRUE]],
+      'uuid' => [(object) ['value' => $this->configFactory->get('system.site')->get('uuid')]],
+      'weight' => [(object) ['value' => 0]],
+    ];
+
+    // Add local menu to the api response.
+    $apiResponse->data->{$project_name} = $site_data;
+
+    return new ResourceResponse(json_decode(json_encode($apiResponse->data), TRUE), 200);
+  }
+
+  /**
+   * Create menu tree from local main menu.
+   *
+   * @return array
+   *   Menu tree.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function getTreeFromMainMenu(): array {
+    $menuId = 'main';
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
```

Comment:
Tää näyttäis olevan identtinen `MenuUpdater::syncMenu` metodin kanssa (miinus lähetys osuus). Tälle vois tehä jonkun oman servicen, esim. `MainMenuBuilder` jota molemmat käyttää main menun buildaamiseen.